main.tf

```diff
@@ -266,6 +266,11 @@ resource "aws_launch_configuration" "gitlab_runner_instance" {
   instance_type        = var.instance_type
   spot_price           = var.runner_instance_spot_price
   iam_instance_profile = aws_iam_instance_profile.instance.name
+  root_block_device {
```

Comment:
This trigger a recreate as you mention which can be avoid by terraform 0.12 dynamic block. I have tested the following config.

```
  dynamic "root_block_device" {
    for_each = [var.runner_root_block_device]
    content {
      delete_on_termination = lookup(root_block_device.value, "delete_on_termination", true)
      volume_type           = lookup(root_block_device.value, "volume_type", "gp2")
      volume_size           = lookup(root_block_device.value, "volume_size", 8)
      iops                  = lookup(root_block_device.value, "iops", null)
    }
  }

```

And as variable add:
```
variable "runner_root_block_device" {
  type    = map(string)
  default = {}
}
```

Now you can set the root block device as following where you declare the module:
```
module "runner" {
   ....
  root_block_device = {
    volume_type = "gp2"
    volume_type = 8
  }
```

Reply:
Good idea! See follow-up commits please.